test/test_autograd.py

```diff
@@ -5761,6 +5773,134 @@ def test_saved_variable_packing_unpacking_did_not_save_original(self):
+            with self.assertRaisesRegex(RuntimeError, "Setting default hooks but they have already been set. "):
+                torch.autograd.graph.set_saved_tensors_default_hooks(lambda x: x, lambda x: x)
+                torch.autograd.graph.set_saved_tensors_default_hooks(lambda x: x, lambda x: x)
+        finally:
+            torch.autograd.graph.reset_saved_tensors_default_hooks()
+
+    def test_saving_variable_to_disk(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            def pack(x):
+                name = os.path.join(tmp_dir, str(uuid.uuid4()))
+                torch.save(x, name)
+                return name
+
+            def unpack(name):
+                return torch.load(name)
+
+            try:
+                torch.autograd.graph.set_saved_tensors_default_hooks(pack, unpack)
+                a = torch.ones(5, requires_grad=True)
+                y = a * a
+                self.assertEqual(a, y.grad_fn._saved_self)
+
+                y.sum().backward()
+                self.assertEqual(2 * a, a.grad)
+            finally:
+                torch.autograd.graph.reset_saved_tensors_default_hooks()
+
+    def test_default_saved_variable_hooks_double_backward(self):
+        try:
+            torch.autograd.graph.set_saved_tensors_default_hooks(lambda x: x, lambda x: x)
+            a = torch.randn(5, requires_grad=True)
+            y = a ** 3
+            s = torch.sum(y)
+            g, = torch.autograd.grad(s, (a, ), create_graph=True)
+            g.sum().backward()
+            self.assertEqual(6 * a, a.grad)
+        finally:
+            torch.autograd.graph.reset_saved_tensors_default_hooks()
+
+        try:
+            torch.autograd.graph.set_saved_tensors_default_hooks(lambda x: 2 * x, lambda x: x)
+            a = torch.randn(5, requires_grad=True)
+            y = a ** 3
+            s = torch.sum(y)
+        finally:
+            torch.autograd.graph.reset_saved_tensors_default_hooks()
+        g, = torch.autograd.grad(s, (a, ), create_graph=True)
+        g.sum().backward()
+        self.assertEqual(6 * 2 * a, a.grad)
+
+        a = torch.randn(5, requires_grad=True)
+        y = a ** 3
+        s = torch.sum(y)
+        try:
+            torch.autograd.graph.set_saved_tensors_default_hooks(lambda x: 2 * x, lambda x: x)
+            g, = torch.autograd.grad(s, (a, ), create_graph=True)
+            g.sum().backward()
+            # factor 4 because pow_backward is grad * (exp * self.pow(exp - 1))
+            self.assertEqual(6 * 4 * a, a.grad)
+        finally:
+            torch.autograd.graph.reset_saved_tensors_default_hooks()
+
+        try:
+            torch.autograd.graph.set_saved_tensors_default_hooks(lambda x: 2 * x, lambda x: x)
+            a = torch.randn(5, requires_grad=True)
+            y = a ** 3
+            s = torch.sum(y)
+            g, = torch.autograd.grad(s, (a, ), create_graph=True)
+            g.sum().backward()
+            self.assertEqual(6 * 8 * a, a.grad)
```

Comment:
The factor of 8 is from... `y` getting saved, `a` getting saved, and a `grad` being saved? (Would be good to include in a comment, otherwise future readers are going to have to think a bit)

Reply:
Actually, `y` is not saved. This test block is the combination of the 2 above. 
In the first block, `a` is saved in `y.grad_fn._saved_self`.
We went over what happens in the second block in https://github.com/pytorch/pytorch/pull/61834#discussion_r675200759.
In the third block, we combine both, so 4 * 2 = 8.
In that sense, `a` is saved twice.

Reply:
Oh I see, thanks for walking through that. This would definitely be nice to have a comment in the code here